src/modules/Form/index.tsx

```diff
@@ -0,0 +1,199 @@
+
+type Props = {
+  componentName: string;
+  location: string;
+};
+
+type State = {
+  isShinyField: Pick<CommonFieldType, 'inputRef'>;
+  pokemons: Array<CustomPokemon & { id: string }>;
+  successMessage: RefObject<HTMLSpanElement>;
+  birthdayField: CommonFieldType;
+  consentField: CommonFieldType;
+  genderField: GenderFieldType;
+  avatarField: CommonFieldType;
+  typeField: SelectFieldType;
+  nameField: CommonFieldType;
+  initialEnter: boolean;
+  totalErrors: number;
+};
+
+export class Form extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      pokemons: [],
+      totalErrors: 0,
+      initialEnter: true,
+      successMessage: createRef(),
+      birthdayField: {
+        inputRef: createRef(),
+        errorRef: createRef(),
+      },
+      consentField: {
+        inputRef: createRef(),
+        errorRef: createRef(),
+      },
+      isShinyField: {
+        inputRef: createRef(),
+      },
+      genderField: {
+        femaleInputRef: createRef(),
+        maleInputRef: createRef(),
+        errorRef: createRef(),
+      },
+      avatarField: {
+        inputRef: createRef(),
+        errorRef: createRef(),
+      },
+      typeField: {
+        selectRef: createRef(),
+        errorRef: createRef(),
+      },
+      nameField: {
+        inputRef: createRef(),
+        errorRef: createRef(),
+      },
+    };
+  }
+
+  clearError = (errorRef: RefObject<HTMLSpanElement>) => {
+    if (this.state.initialEnter) this.setState({ initialEnter: false });
+    if (errorRef.current?.style.display === 'block') {
+      this.setState({ totalErrors: this.state.totalErrors - 1 });
+      errorRef.current.style.display = 'none';
+    }
+  };
+
+  clearFields = () => {
+    const {
+      birthdayField,
+      isShinyField,
+      consentField,
+      genderField,
+      avatarField,
+      nameField,
+      typeField,
+    } = this.state;
+    avatarField.inputRef.current && (avatarField.inputRef.current.value = '');
+    birthdayField.inputRef.current && (birthdayField.inputRef.current.value = '');
+    consentField.inputRef.current && (consentField.inputRef.current.checked = false);
+    genderField.maleInputRef.current && (genderField.maleInputRef.current.checked = false);
+    genderField.femaleInputRef.current && (genderField.femaleInputRef.current.checked = false);
+    isShinyField.inputRef.current && (isShinyField.inputRef.current.checked = false);
+    typeField.selectRef.current && (typeField.selectRef.current.value = '');
+    nameField.inputRef.current && (nameField.inputRef.current.value = '');
+  };
+
+  validateGenderField = () => {
+    const { errorRef, femaleInputRef, maleInputRef } = this.state.genderField;
+    if (femaleInputRef.current?.checked) return 'female';
+    else if (maleInputRef.current?.checked) return 'male';
+    else errorRef?.current && (errorRef.current.style.display = 'block');
+  };
+
+  validateConsentField = () => {
```

Comment:
If there will be a form with 1 million fields you will make separate function for each of it even with the same validation?
hint: You can understand which input you are dealing with by the attribute `name`